tokio/src/util/linked_list.rs

```diff
@@ -128,6 +132,24 @@ impl<T: Link> LinkedList<T> {
         true
     }
 
+    pub(crate) fn is_linked(&self, node: &T::Handle) -> bool {
```

Comment:
It is important to note that this function can return a false positive / negative depending on what this function is actually doing.

Does `is_linked` mean that the node is linked in **any** linked list or in **this** one. The function implies the latter. If this is the case, then if the node is linked in a **different**, non-empty, list, the function will return `true` even though it should return `false`.

I would document this as well as a big warning / disclaimer. The function is not `unsafe` but any `unsafe` code block that relies on specific behavior must be cautious.

Reply:
Yeah, this is a good point. Originally, this was a function implemented for `Links` that just checked if the next & prev pointers were `None`, so it was clearer that it tested if the node was linked with _any_ list. However, testing _just_ that method caused issues when the node in question was the only node in a list.

It might be better to move this back to the node type, and have the test in the semaphore be
```rust
if node.is_linked() || waiters.head == node { 
    // ...
```
or something. That way, the API here only implies that the method tests if the node is linked with other nodes, and the other checks are specific to the semaphore.

Reply:
hopefully 3b4f41638332c61f7adfc7f3180a9eb2d45fa4aa will make this a little harder to misuse?